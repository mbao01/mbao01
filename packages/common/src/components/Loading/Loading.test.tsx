import { describe, expect, it } from "vitest";
import { Loading } from "./";
import { render, screen } from "@testing-library/react";

describe("Loading", () => {
  it("shows a tiny colored spinner", () => {
    const { asFragment } = render(
      <Loading
        color="primary"
        size="xs"
        variant="spinner"
        data-testid="loading"
      />
    );

    expect(screen.getByTestId("loading")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each(["spinner", "dots", "ring", "ball", "bars", "infinity"] as const)(
    "has %s variant",
    (variant) => {
      const { asFragment } = render(
        <Loading variant={variant} data-testid="loading" />
      );

      expect(screen.getByTestId("loading")).toHaveClass(`loading-${variant}`);
      expect(asFragment()).toMatchSnapshot();
    }
  );

  it.each([
    { size: "xs", name: "tiny" },
    { size: "sm", name: "small" },
    { size: "md", name: "base" },
    { size: "lg", name: "large" },
  ] as const)("has $name ($size) loading", ({ size }) => {
    const { asFragment } = render(
      <Loading size={size} data-testid="loading" />
    );

    expect(screen.getByTestId("loading")).toHaveClass(`loading-${size}`);
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    "primary",
    "secondary",
    "accent",
    "neutral",
    "info",
    "success",
    "warning",
    "error",
  ] as const)("has %s color", (intent) => {
    const { asFragment } = render(
      <Loading intent={intent} data-testid="loading" />
    );

    expect(screen.getByTestId("loading")).toHaveClass(`text-${intent}`);
    expect(asFragment()).toMatchSnapshot();
  });
});
