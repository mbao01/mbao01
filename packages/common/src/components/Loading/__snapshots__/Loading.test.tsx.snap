// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > has 'base' ('md') loading 1`] = `
<DocumentFragment>
  <span
    class="loading loading-md"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has 'large' ('lg') loading 1`] = `
<DocumentFragment>
  <span
    class="loading loading-lg"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has 'small' ('sm') loading 1`] = `
<DocumentFragment>
  <span
    class="loading loading-sm"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has 'tiny' ('xs') loading 1`] = `
<DocumentFragment>
  <span
    class="loading loading-xs"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has accent color 1`] = `
<DocumentFragment>
  <span
    class="loading text-accent-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has ball variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-ball"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has bars variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-bars"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has dots variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-dots"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has error color 1`] = `
<DocumentFragment>
  <span
    class="loading text-error-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has infinity variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-infinity"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has info color 1`] = `
<DocumentFragment>
  <span
    class="loading text-info-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has neutral color 1`] = `
<DocumentFragment>
  <span
    class="loading text-neutral-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has primary color 1`] = `
<DocumentFragment>
  <span
    class="loading text-primary-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has ring variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-ring"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has secondary color 1`] = `
<DocumentFragment>
  <span
    class="loading text-secondary-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has spinner variant 1`] = `
<DocumentFragment>
  <span
    class="loading loading-spinner"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has success color 1`] = `
<DocumentFragment>
  <span
    class="loading text-success-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > has warning color 1`] = `
<DocumentFragment>
  <span
    class="loading text-warning-content"
    data-testid="loading"
  />
</DocumentFragment>
`;

exports[`Loading > shows a tiny colored spinner 1`] = `
<DocumentFragment>
  <span
    class="loading loading-spinner loading-xs"
    color="primary"
    data-testid="loading"
  />
</DocumentFragment>
`;
